From hackermail Thu Jan  1 00:00:00 1970
From: SeongJae Park <sj@kernel.org>
To: Andrew Morton <akpm@linux-foundation.org>
Cc: SeongJae Park <sj@kernel.org>, damon@lists.linux.dev, kernel-team@meta.com, linux-kernel@vger.kernel.org, linux-mm@kvack.org
Message-Id: <20250712195016.151108-9-sj@kernel.org>
In-Reply-To: <20250712195016.151108-1-sj@kernel.org>
Date: Sat, 12 Jul 2025 12:50:10 -0700
Subject: [PATCH 08/14] mm/damon/core: do not call ops.cleanup() when destroying targets

damon_operations.cleanup() is documented to be called for kdamond
termination, but also being called for targets destruction, which is
done for any damon_ctx destruction.  Nobody is using the callback for
now, though.  Remove the cleanup() call under the destruction.

Signed-off-by: SeongJae Park <sj@kernel.org>
Link: https://patch.msgid.link/20250712195016.151108-9-sj@kernel.org
---
 mm/damon/core.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index ffb87497dbb5..b82a838b5a0e 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -550,11 +550,6 @@ static void damon_destroy_targets(struct damon_ctx *ctx)
 {
 	struct damon_target *t, *next_t;
 
-	if (ctx->ops.cleanup) {
-		ctx->ops.cleanup(ctx);
-		return;
-	}
-
 	damon_for_each_target_safe(t, next_t, ctx)
 		damon_destroy_target(t);
 }
-- 
2.39.5